tests/test_scenario/test_moving_push_notifications.py

```diff
@@ -0,0 +1,505 @@
+                        )
+                    except Exception as e:
+                        print(f"Error reading push notification: {e}")
+                        break
+
+        finally:
+            # Release the connection back to the pool
+            for conn in connections_to_check:
+                try:
+                    pool.release(conn)
+                except Exception as e:
+                    print(f"Error releasing connection: {e}")
+
+    def _monitor_maintenance_state_during_operation(
+        self, action_id: str, operation_name: str, timeout: int = 120
+    ):
+        """Monitor maintenance state changes while a Redis Enterprise operation
+        is running."""
+
+        start_time = time.time()
+        check_interval = 3  # Check more frequently during operations
+        maintenance_states_found = set()
+        print(
+            f"Starting maintenance state monitoring for {operation_name} "
+            f"operation {action_id}"
+        )
+
+        while time.time() - start_time < timeout:
+            try:
+                self._check_for_push_notifications()
+                pool = self.redis_client.connection_pool
+
+                # Check available connections
+                if hasattr(pool, "_available_connections"):
+                    for conn in pool._available_connections:
+                        state = getattr(
+                            conn, "maintenance_state", MaintenanceState.NONE
+                        )
+                        if state != MaintenanceState.NONE:
+                            maintenance_states_found.add(state)
+
+                print(f"Maintenance states found: {maintenance_states_found}")
+
+                # Check operation status first
+                status_result = self.fault_injector.get_action_status(action_id)
+                operation_status = status_result.get("status", "unknown")
+
+                completed_statuses = [
+                    "failed",
+                    "finished",
+                ]
+                if operation_status in completed_statuses:
+                    print(
+                        f"Operation {action_id} completed with status: "
+                        f"{operation_status}"
+                    )
+                    break
+
+                time.sleep(check_interval)
+
+            except Exception as e:
+                print(f"Error checking operation status: {e}")
+                time.sleep(check_interval)
+        else:
+            print(f"Timeout waiting for {operation_name} operation {action_id}")
+
+        print(
+            f"Maintenance states detected during {operation_name}: "
+            f"{maintenance_states_found}"
+        )
+        return maintenance_states_found
+
+    def _get_cluster_nodes_info(self) -> Dict[str, Any]:
+        """Get cluster nodes information from Redis Enterprise."""
+        try:
+            # Use rladmin status to get node information
+            action = ActionRequest(
```

Comment:
The same achievable with REST API

https://redis.io/docs/latest/operate/rs/references/rest-api/requests/nodes/#get-all-nodes